File: CzhERP.Application/Entity/PurReturnItem.cs
namespace CzhERP.Application.Entity;

[SugarTable("Pur_ReturnItem", TableDescription = "采购退货明细表")]
public class PurReturnItem : EntityBaseId
{
    [SugarColumn(ColumnDescription = "退货单ID", IsNullable = false)]
    public long ReturnId { get; set; }

    [SugarColumn(ColumnDescription = "入库明细ID")]
    public long? InboundItemId { get; set; }

    [SugarColumn(ColumnDescription = "物料ID", IsNullable = false)]
    public long MaterialId { get; set; }

    [SugarColumn(ColumnDescription = "物料编码", Length = 50, IsNullable = false)]
    public string MaterialCode { get; set; }

    [SugarColumn(ColumnDescription = "物料名称", Length = 100, IsNullable = false)]
    public string MaterialName { get; set; }

    [SugarColumn(ColumnDescription = "规格型号", Length = 100)]
    public string? Spec { get; set; }

    [SugarColumn(ColumnDescription = "单位名称", Length = 50, IsNullable = false)]
    public string UnitName { get; set; }

    [SugarColumn(ColumnDescription = "退货数量", IsNullable = false)]
    public decimal ReturnQty { get; set; }

    [SugarColumn(ColumnDescription = "单价", IsNullable = false)]
    public decimal Price { get; set; }

    [SugarColumn(ColumnDescription = "金额", DefaultValue = "0")]
    public decimal Amount { get; set; } = 0;

    [SugarColumn(ColumnDescription = "退货原因", Length = 200)]
    public string? Reason { get; set; }

    [SugarColumn(ColumnDescription = "排序", DefaultValue = "0")]
    public int SortOrder { get; set; } = 0;

    [Navigate(NavigateType.OneToOne, nameof(ReturnId))]
    public PurReturn? Return { get; set; }

    [Navigate(NavigateType.OneToOne, nameof(InboundItemId))]
    public PurInboundItem? InboundItem { get; set; }
}

File: Web/src/api/purchase/purReturnItem.ts
import {useBaseApi} from '/@/api/base';

// 采购退货明细表接口服务
export const usePurReturnItemApi = () => {
	const baseApi = useBaseApi("purReturnItem");
	return {
		// 分页查询采购退货明细表
		page: baseApi.page,
		// 查看采购退货明细表详细
		detail: baseApi.detail,
		// 新增采购退货明细表
		add: baseApi.add,
		// 更新采购退货明细表
		update: baseApi.update,
		// 删除采购退货明细表
		delete: baseApi.delete,
		// 批量删除采购退货明细表
		batchDelete: baseApi.batchDelete,
		// 导出采购退货明细表数据
		exportData: baseApi.exportData,
		// 导入采购退货明细表数据
		importData: baseApi.importData,
		// 下载采购退货明细表数据导入模板
		downloadTemplate: baseApi.downloadTemplate,
	}
}

// 采购退货明细表实体
export interface PurReturnItem {
	// 主键Id
	id: number;
	// 退货单ID
	returnId?: number;
	// 入库明细ID
	inboundItemId: number;
	// 物料ID
	materialId?: number;
	// 物料编码
	materialCode?: string;
	// 物料名称
	materialName?: string;
	// 规格型号
	spec: string;
	// 单位名称
	unitName?: string;
	// 退货数量
	returnQty?: number;
	// 单价
	price?: number;
	// 金额
	amount?: number;
	// 退货原因
	reason: string;
	// 排序
	sortOrder?: number;
}